[WCiOS17] Use `registerForTraitChanges` in refund views

iamgabrielma opened the pull request:
Part of WOOMOB-1003

## Description
This PR addresses `traitCollectionDidChange' was deprecated in iOS 17.0: Use the trait change registration APIs declared in the UITraitChangeObservable protocol` warning in the refund flow. 

In the new API (iOS17+), we register the traits we care about when initializing the view and the system will call the handler when traits change, rather than doing a subclass override.

## Testing information
- Add a breakpoint to one of the functions that is called on category size updates, example on  `IssueRefundViewController.configureHeaderStackView()` in line 187, and `RefundItemTableViewCell.adjustItemsStackViewAxis()` line 98
- Go to Orders, find a completed order, tap on `Refund`, observe the breakpoint is called. 
- Increase or decrease the category size (`cmd` + `option` + `+`) to trigger the trait observation change, note the breakpoint is triggered again

<img width="885" height="526" alt="Screenshot 2025-09-04 at 11 13 41" src="https://github.com/user-attachments/assets/b28f32c1-e570-4618-bc3b-ad229d664051" />

iamgabrielma committed:
Use registerForTraitChanges for refund views

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
update refund’s shipping details trait observer

iamgabrielma committed:
update POSFullScreenCover to new onchange API

iamgabrielma committed:
Update usage of DDLogError

Passing a string as such is currently deprecated, we need to pass it interpolated through: https://github.com/CocoaLumberjack/CocoaLumberjack/discussions/1394

iamgabrielma committed:
update ordertableviewcell to use registerForTraitChanges API

iamgabrielma committed:
update FilteredOrdersHeaderBar to use registerForTraitChanges API

iamgabrielma committed:
lint

iamgabrielma committed:
periphery lint

iamgabrielma committed:
typo

RafaelKayumov reviewed (approved):
LGTM. Changes look clean to me. Approving to unblock merging before code freeze.

Though I have an observation. Tested on iOS 18.4 iPhone 16 simulator:
- Registered trait changes observer worked fine for `IssueRefundViewController`. Toggling accessibility text sizes triggered the registered closure.
- Registered trait changes observer was never called for `RefundItemTableViewCell`. Tried both text size toggling and rotating the device - no luck. I Returned back the `traitCollectionDidChange` override in `RefundItemTableViewCel` and it got triggered on device rotation.

I wonder if it's enough to only register `UITraitPreferredContentSizeCategory` trait? There is plenty of predefined UITraitDefinition. For example `UITraitUserInterfaceIdiom` or `UITraitVerticalSizeClass` might be worth including into `registerForTraitChanges` to have closer behaviour to old `traitCollectionDidChange`

iamgabrielma committed:
[WCiOS17] Use `registerForTraitChanges` in order views, and miscellanea warnings around POS/variations (#16077)

iamgabrielma committed:
move trait detection later to the lifecycle

iamgabrielma committed:
expand traits registration for refund and shipping views

iamgabrielma committed:
lint

iamgabrielma commented:
Thanks for the review and testing! 

> Though I have an observation. Tested on iOS 18.4 iPhone 16 simulator:
> * Registered trait changes observer was never called for `RefundItemTableViewCell`. Tried both text size toggling and rotating the device - no luck. I Returned back the `traitCollectionDidChange` override in `RefundItemTableViewCel` and it got triggered on device rotation.

Ah! I saw something off as well on a separate PR on different iOS version. For some reason, depending on the view is not enough to start observing on `awakeFromNib`, seems to be too early on the lifecycle, and moving it to `willMove` fixes it. I'm still not sure why and seems to need to be tested per case basis. I'll see if I find a common denominator while fixing the rest.

Updated: https://github.com/woocommerce/woocommerce-ios/pull/16076/commits/24697637c41dac21b87948d0395ea13c3df44830

> I wonder if it's enough to only register `UITraitPreferredContentSizeCategory` trait? There is plenty of predefined UITraitDefinition. For example `UITraitUserInterfaceIdiom` or `UITraitVerticalSizeClass` might be worth including into `registerForTraitChanges` to have closer behaviour to old `traitCollectionDidChange`

Good point, updated to expand traits here: https://github.com/woocommerce/woocommerce-ios/pull/16076/commits/89fefde10717be97cd25ad6ffedc23a499d6b604
